scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -141,6 +141,21 @@ def test_reproduce(self):
         assert_equal(res1.x, res2.x)
         assert_equal(res1.x, res3.x)
 
+    @pytest.mark.skipif(np.__version__ < '1.17',
+                        reason='Generator not available for numpy, < 1.17')
```

Comment:
Minor point, but I believe the robust way to do version comparisons now is to use PEP440 machinery, as recently described by @peterbell10 in #11290.

Reply:
made changes